controllers/configuration/backend/backend.go

```diff
@@ -0,0 +1,335 @@
+	"azurerm": {
+		"client_certificate_path": "ClientCertificateSecret",
+	},
+	"consul": {
+		"ca_file":   "CAFileSecret",
+		"cert_file": "CertFileSecret",
+		"key_file":  "KeyFileSecret",
+	},
+	"etcdv3": {
+		"cacert_path": "CacertSecret",
+		"cert_path":   "CertSecret",
+		"key_path":    "KeySecret",
+	},
+	"gcs": {
+		"credentials": "CredentialsSecret",
+	},
+	"kubernetes": {
+		"config_path":  "ConfigSecret",
+		"config_paths": "ConfigSecrets",
+	},
+	"oss": {
+		"shared_credentials_file": "SharedCredentialsSecret",
+	},
+	"s3": {
+		"shared_credentials_file": "SharedCredentialsSecret",
+	},
+	"swift": {
+		"cacert_file": "CacertSecret",
+		"cert":        "CertSecret",
+		"key":         "KeySecret",
+	},
+}
+
+var backendTypes = []string{
+	"remote", "artifactory", "azurerm", "consul", "cos", "etcd", "etcdv3", "gcs", "http", "kubernetes",
+	"manta", "oss", "pg", "s3", "swift",
+}
+
+var backendTF = `
+terraform {
+  backend "kubernetes" {
+    secret_suffix     = "{{.SecretSuffix}}"
+    in_cluster_config = {{.InClusterConfig}}
+    namespace         = "{{.Namespace}}"
+  }
+}
+`
+
+// Conf contains all the backend information in the meta
+type Conf struct {
+	BackendType string
+	HCL         string
+	// UseCustom indicates whether to use custom kubernetes backend
+	UseCustom bool
+	// Secrets describes which secret and which key in the secret should be mounted to the executor pod
+	Secrets []*ConfSecretSelector
+}
+
+// ConfSecretSelector describes which secret should be mounted to the executor pod
+type ConfSecretSelector struct {
+	// Name is the name of the secret which will be mounted to a pod when running the job
+	Name string
+	// Path is a relative path, indicates which path the secret should be mounted to
+	Path string
+	// Keys are the keys that contains the file content in secret
+	Keys []string
+}
+
+type backendVars struct {
+	SecretSuffix    string
+	InClusterConfig bool
+	Namespace       string
+}
+
+// renderTemplate renders Backend template
+func renderTemplate(backend *v1beta2.Backend, namespace string) (string, error) {
+	tmpl, err := template.New("backend").Funcs(template.FuncMap(sprig.FuncMap())).Parse(backendTF)
+	if err != nil {
+		return "", err
+	}
+
+	templateVars := backendVars{
+		SecretSuffix:    backend.SecretSuffix,
+		InClusterConfig: backend.InClusterConfig,
+		Namespace:       namespace,
+	}
+	var wr bytes.Buffer
+	err = tmpl.Execute(&wr, templateVars)
+	if err != nil {
+		return "", err
+	}
+	return wr.String(), nil
+}
+
+func checkBackendTypeValid(backendType string) bool {
+	for _, v := range backendTypes {
+		if v == backendType {
+			return true
+		}
+	}
+	return false
+}
+
+// ParseConfigurationBackend returns backend hcl string, backend type, useCustom, secretRef list and error
+func ParseConfigurationBackend(configuration *v1beta2.Configuration, terraformBackendNamespace string) (*Conf, error) {
+	backend := configuration.Spec.Backend
+
+	switch {
+
+	case backend != nil && len(backend.Inline) > 0:
```

Comment:
Check backend is nil before the switch, then you can remove the check from all cases.

Reply:
But the `backend == nil` condition is the default condition and some other conditions like `backend == nil && backend.backendType == ""` will fallthrough to this codition. I think putting it the end is better.